tests/install_tests.rs

```diff
@@ -33,3 +39,106 @@ async fn install_run_python() {
     assert!(result.success());
     assert_eq!(result.stdout().trim(), "Python 3.11.0");
 }
+
+#[tokio::test]
+async fn init_creates_project_manifest() {
+    let tmp_dir = TempDir::new().unwrap();
+
+    // Run the init command
+    pixi::cli::init::execute(pixi::cli::init::Args {
+        path: tmp_dir.path().to_path_buf(),
+    })
+    .await
+    .unwrap();
+
+    // There should be a loadable project manifest in the directory
+    let project = Project::load(&tmp_dir.path().join(pixi::consts::PROJECT_MANIFEST)).unwrap();
+
+    // Default configuration should be present in the file
+    assert!(!project.name().is_empty());
+    assert_eq!(project.version(), &Version::from_str("0.1.0").unwrap());
+}
+
+/// This is a test to check that creating incremental lock files works.
+///
+/// It works by using a fake channel that contains two packages: `foo` and `bar`. `foo` depends on
+/// `bar` so adding a dependency on `foo` pulls in `bar`. Initially only version `1` of both
+/// packages is added and a project is created that depends on `foo >=1`. This select `foo@1` and
+/// `bar@1`.
+/// Next, version 2 for both packages is added and the requirement in the project is updated to
+/// `foo >=2`, this should then select `foo@1` but `bar` should remain on version `1` even though
+/// version `2` is available. This is because `bar` was previously locked to version `1` and it is
+/// still a valid solution to keep using version `1` of bar.
+#[tokio::test]
+async fn test_incremental_lock_file() {
+    let mut package_database = PackageDatabase::default();
+
+    // Add a package `foo` that depends on `bar` both set to version 1.
+    package_database.add_package(Package::build("bar", "1").finish());
+    package_database.add_package(
+        Package::build("foo", "1")
+            .with_dependency("bar >=1")
+            .finish(),
+    );
+
+    // Write the repodata to disk
+    let channel_dir = TempDir::new().unwrap();
+    package_database
+        .write_repodata(channel_dir.path())
+        .await
+        .unwrap();
+
+    let mut pixi = PixiControl::new().unwrap();
+
+    // Create a new project
+    pixi.init().await.unwrap();
+
+    // Set the channel to something we created
+    pixi.set_channel(Url::from_directory_path(channel_dir.path()).unwrap())
```

Comment:
Shouldn't we also add this option to the command line interface? Maybe even something like an add channel option or as argument to init?

Reply:
Yeah ideally we'd add this to the `init` command IMHO. I wasn't sure whether to do this in this PR though.

Reply:
I added channels to the init command and modified PixiControl to allow you to use it.